### PR TITLE
Knutt Morris Prat

### PR DESCRIPTION
**Pull request by salonishah01**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

C code for KMP Algorithm

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1323

### CODE REVIEW

File: C/ds/KnuttMorrisPratAlgo.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                j = 0;
+            }
+        }
+    }
+    printf("No substring match found in the string.\n");
+ 
+    return 0;
+}
+
+/*
```
Comment (atarax665) [inline C/ds/KnuttMorrisPratAlgo.c]: Add time/space complexity of the algorithm.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: C/ds/KnuttMorrisPratAlgo.c (hunk 1 above)
Comment (salonishah01) [reply C/ds/KnuttMorrisPratAlgo.c:57]: @atarax665 Added.

Comment (salonishah01) [reply C/ds/KnuttMorrisPratAlgo.c]: @atarax665 Added.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Use comments wherever needed to improve readability of your code. Also, move the file to ```cp folder``` and make the changes in README accordingly.

**Comment by salonishah01** [comment]
@atarax665 Moved to cp folder and added comments.-

**Review by iamrajiv — APPROVED** [review]

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Pull request merged**